controllers/dns_helper.go

```diff
@@ -113,3 +115,35 @@ func (g GatewayWrapper) GetAddresses() []builder.TargetAddress {
 	}
 	return addresses
 }
+
+func (g *GatewayWrapper) SetValidStatusAddresses(p *v1alpha1.DNSPolicy) error {
+	newAddresses := []gatewayapiv1.GatewayStatusAddress{}
+	for _, address := range g.Gateway.Status.Addresses {
+		found := false
+		for _, exclude := range p.Spec.ExcludeAddresses {
+			//Only a CIDR will have  / in the address so attempt to parse fail if not valid
+			if strings.Contains(exclude, "/") {
+				_, network, err := net.ParseCIDR(exclude)
+				if err != nil {
+					return fmt.Errorf("could not parse the CIDR from the excludeAddresses field %w", err)
+				}
+				ip := net.ParseIP(address.Value)
+				// only check addresses that are actually IPs
+				if ip != nil && network.Contains(ip) {
+					found = true
+					break
+				}
+			}
+			if exclude == address.Value {
+				found = true
+				break
+			}
+		}
+		if !found {
+			newAddresses = append(newAddresses, address)
+		}
+	}
+	// setting this in memory only wont be saved to actual gateway
+	g.Status.Addresses = newAddresses
+	return nil
+}
```

Comment:
Do you not need to call this somewhere, or am i just not seeing where you are doing that?

Reply:
Another option to this would be to just add the excluded addresses array to the gateway wrapper `SetExcludeAddresses(excludeAddresses []string)` and filter out excluded addresses as part of `GetAddresses()` rather than modifying the gateway status itself.

Reply:
might have been dropped in the rebase..